common/fs/src/cache/mod.rs

```diff
@@ -35,6 +35,46 @@ type WatchDescriptors = HashMap<WatchDescriptor, Vec<EntryKey>>;
 pub type EntryKey = DefaultKey;
 
 type EntryMap<T> = SlotMap<EntryKey, RefCell<entry::Entry<T>>>;
+type FsResult<T> = Result<T, Error>;
+
+#[derive(std::fmt::Debug)]
+pub enum Error {
```

Comment:
Very nice.

Would you mind redoing this using thiserror's proc macro? You can see examples in the k8s middleware. It derives std::error::Error and various other convenience things.

It should look something like
``` rust
use thiserror::Error;

#[derive(Clone, Debug, Error)]
pub enum K8sError {
    #[error("pod missing {0}")]
    PodMissingMetaError(&'static str),
    #[error("failed to initialize kubernetes middleware {0}")]
    InitializationError(String),
}


#[derive(Clone, Debug, Error)]
pub enum Error {
    #[error("error watching: {0:?} {1:?}")]
    Watch(PathBuf, io::Error),
    #[error("got event for untracked watch descriptor: {0:?}")]
    WatchEvent(WatchDescriptor),
    #[error("the inotify event queue has overflowed and events have presumably been lost")]
    WatchOverflow,
    #[error("unexpected existing entry")]
    Existing,
    #[error("failed to find entry")]
    Lookup,
    #[error("failed to find parent entry")]
    ParentLookup,
    #[error("parent should be a directory")]
    ParentNotValid,
    #[error("path is not valid")]
    PathNotValid,
    #[error("encountered errors when inserting recursively: {0:?}")]
    InsertRecursively(Vec<Error>),
}
```

Reply:
oh! TIL!

Thanks for pointing it out, I'll update it.